src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/InnerAssignmentCheck.java

```diff
@@ -264,4 +279,19 @@ private static boolean isInContext(DetailAST ast, int[]... contextSet) {
         return found;
     }
 
+    /**
+     * Get ast parent, ignoring token types from {@code skipTokens}.
+     *
+     * @param ast token to get parent
+     * @param skipTokens token types to skip
+     * @return first not ignored parent of ast
+     */
+    private static DetailAST getParent(DetailAST ast, int... skipTokens) {
+        DetailAST result = ast.getParent();
+        while (Arrays.binarySearch(skipTokens, result.getType()) > -1) {
+            result = result.getParent();
```

Comment:
Isn't this bad to be endlessly checking parent after parent?
What happens if we have an anonymous class inside the while loop. If there is an assignment in the anonymous class, won't this break out of the anonymous class and see the while loop and falsely pick that as the parent?
Something to that nature.

Reply:
Can you please provide code example?

Reply:
I am not sure where this code is in terms of the tree it is looking at, but I am thinking of something along the lines:
````
while (true && new Test() {
    public boolean test() {
        boolean result;
        result = true;
        return result;
    }.getResult()) {
  // example
}
````

Reply:
I still do not understand. 
Before it was looking on direct parent and checked if it matches sequence. Right now it does exactly the same thing for other cases, but for while idiom it skips boolean operators.
In your code there are no boolean operators in assignment parents.

Reply:
I will have to play with the code more then I can probably explain better.